app/src/test/java/com/simplemobiletools/calculator/MainActivityTest.kt

```diff
@@ -25,146 +23,192 @@ class MainActivityTest {
     @Test
     fun addSimpleDigit() {
         activity.calc.addDigit(2)
-        assertEquals("2", getDisplayedNumber())
+        assertEquals("2", activity.getResult())
     }
 
     @Test
     fun removeLeadingZero() {
         activity.calc.addDigit(0)
         activity.calc.addDigit(5)
-        assertEquals("5", getDisplayedNumber())
+        assertEquals("5", activity.getResult())
     }
 
     @Test
     fun additionTest() {
-        val res = calcResult(-1.2, PLUS, 3.4)
-        assertEquals("2.2", res)
-        checkFormula("-1.2+3.4")
+        activity.calc.addDigit(5)
+        activity.calc.handleOperation("plus")
+        activity.calc.addDigit(4)
+        activity.calc.handleEquals()
+        assertEquals("9", activity.getResult())
+        checkFormula("5+4")
     }
 
     @Test
-    fun subtractionTest() {
-        val res = calcResult(7.8, MINUS, 2.5)
-        assertEquals("5.3", res)
-        checkFormula("7.8-2.5")
+    fun subtractTest() {
+        activity.calc.addDigit(5)
+        activity.calc.handleOperation("minus")
+        activity.calc.addDigit(4)
+        activity.calc.handleEquals()
+        var result  = activity.getResult()
+
+        assertEquals("1", result)
+        checkFormula("5-4")
     }
 
     @Test
     fun multiplyTest() {
-        val res = calcResult(-3.2, MULTIPLY, 6.6)
-        assertEquals("-21.12", res)
-        checkFormula("-3.2*6.6")
+        activity.calc.addDigit(5)
+        activity.calc.handleOperation("multiply")
+        activity.calc.addDigit(4)
+        activity.calc.handleEquals()
+        var result  = activity.getResult()
+
+        assertEquals("20", result)
+        checkFormula("5×4")
     }
 
     @Test
     fun divisionTest() {
-        val res = calcResult(18.25, DIVIDE, 5.0)
-        assertEquals("3.65", res)
-        checkFormula("18.25/5")
+        activity.calc.addDigit(6)
+        activity.calc.handleOperation("divide")
+        activity.calc.addDigit(2)
+        activity.calc.handleEquals()
+        var result  = activity.getResult()
+
+        assertEquals("3", result)
     }
 
     @Test
-    fun divisionByZero_returnsZero() {
-        val res = calcResult(6.0, DIVIDE, 0.0)
-        assertEquals("0", res)
-        checkFormula("6/0")
-    }
+    fun divisionByZero_returnsInfinity() {
+        activity.calc.addDigit(6)
+        activity.calc.handleOperation("divide")
+        activity.calc.addDigit(0)
+        activity.calc.handleEquals()
+        var result  = activity.getResult()
+
+        print("result: "+result)
+        assertEquals("∞", result)
```

Comment:
nice. Didn't know my fix would output the infinity sign 😅

Reply:
🤑🤫🤥